compiler/docgen.nim

```diff
@@ -1056,10 +1057,18 @@ proc writeOutput*(d: PDoc, useWarning = false) =
   else:
     template outfile: untyped = d.destFile
     #let outfile = getOutFile2(d.conf, shortenDir(d.conf, filename), outExt, "htmldocs")
-    createDir(outfile.splitFile.dir)
```

Comment:
But look, read this code: We want to create the directory that out**file** resides in. 

Your version: "Use the parentDir of outfile". This is worse.

Reply:
done
(not sure i understand though; if anything the version i wrote seems more efficient; maybe parentDir is lacking in pathutils but it'd have to handle each of AbsoluteFile, AbsoluteDir, RelativeFile, RelativeDir)